plugins/module_utils/dnac.py

```diff
@@ -992,6 +993,182 @@ def check_tasks_response_status(self, response, api_name):
 
         return self
 
+    def update_result(self, status, changed, msg, log_level, data=None):
+        """
+        Update the result of the operation with the provided status, message, and log level.
+        Parameters:
+            - status (str): The status of the operation ("success" or "failed").
+            - changed (bool): Indicates whether the operation caused changes.
+            - msg (str): The message describing the result of the operation.
+            - log_level (str): The log level at which the message should be logged ("INFO", "ERROR", "CRITICAL", etc.).
+            - data (dict, optional): Additional data related to the operation result.
+        Returns:
+            self (object): An instance of the class.
+        Note:
+            - If the status is "failed", the "failed" key in the result dictionary will be set to True.
+            - If data is provided, it will be included in the result dictionary.
+        """
+        # Update the result attributes with the provided values
+        self.status = status
+        self.result["status"] = status
+        self.result["msg"] = msg
+        self.result["changed"] = changed
+
+        # Log the message at the specified log level
+        self.log(msg, log_level)
+
+        # If the status is "failed", set the "failed" key to True
+        if status == "failed":
+            self.result["failed"] = True
+
+        # If additional data is provided, include it in the result dictionary
+        if data:
+            self.result["data"] = data
+
+        return self
+
+    def fail_and_exit(self, msg):
+        """Helper method to update the result as failed and exit."""
+        self.update_result("failed", False, msg, "ERROR")
+        self.check_return_status()
+
+    def log_traceback(self):
+        """
+        Logs the full traceback of the current exception.
+        """
+        # Capture the full traceback
+        full_traceback = traceback.format_exc()
+        
+        # Log the traceback
+        self.log("Traceback: {0}".format(full_traceback), "DEBUG")
+
+    def exit_while_loop(self, start_time, task_id, task_name, response):
+        """
+        Check if the elapsed time exceeds the specified timeout period and exit the while loop if it does.
+        Parameters:
+            - start_time (float): The time when the while loop started.
+            - task_id (str): ID of the task being monitored.
+            - task_name (str): Name of the task being monitored.
+            - response (dict): Response received from the task status check.
+        Returns:
+            bool: True if the elapsed time exceeds the timeout period, False otherwise.
+        """
+        # If the elapsed time exceeds the timeo.,ut period
```

Comment:
Can we say elapsed_time?

    # Calculate the elapsed time
    elapsed_time = time.time() - start_time
    
    # If the elapsed time exceeds the timeout period
    if elapsed_time > self.params.get("dnac_api_task_timeout"):